### PR TITLE
Allow project view for bazel projects

### PR DESCRIPTION
**Pull request by sohaibiftikhar**
Sometimes we don't need analysis against the entire project but parts of the project are enough. In that case, using a reduced target is more efficient.

### CODE REVIEW

File: src/main/java/org/javacs/InferConfig.java [hunk 1]
```diff
@@ -283,14 +283,30 @@ private Path bazelOutputBase(Path bazelWorkspaceRoot) {
...
+    private String projectView(Path bazelWorkspaceRoot) {
+        Path projectFile = bazelWorkspaceRoot.resolve(".project");
+        String projectTargets = "...";
+        LOG.info("Searching for project file in " + projectFile.toString());
+        try {
+            if ( Files.exists(projectFile) ) {
+                projectTargets = String.join(" ", Files.readAllLines(projectFile));
+            }
+            LOG.info("Found projectTargets: " + projectTargets);
+        } catch (IOException exception) {
```
Comment (sohaibiftikhar) [inline src/main/java/org/javacs/InferConfig.java:295]: Happy for suggestions on how to handle error here? Just log and let it go? Or does it need to be fatal?

**Comment by georgewfraser** [comment]
I'm a little worried about incorporated something that's fundamentally an intellij feature, not a bazel feature. https://ij.bazel.build/docs/project-views.html

**Comment by sohaibiftikhar** [comment]
I actually added this to my setup when I was setting up code for Bazel (https://bazel.build/basics/getting_started.html) to reduce the load on the lang server (I use vim as an editor). So it does indeed originate from Intellij. But I feel its completely portable to any other editor/lang-server setup.
It works for me quite well. But you can go ahead and close this.

**Comment by georgewfraser** [comment]
Does https://github.com/georgewfraser/java-language-server/pull/161 solve the same problem in a different way?

**Comment by sohaibiftikhar** [comment]
In some ways I suppose. From what I read it still doesn't solve it if I have three subfolders and want two of them included.

**Comment by georgewfraser** [comment]
Yeah that's true, it allows you to choose a single subfolder.

**Review by APA2527 — APPROVED** [review]

**Pull request closed without merging**